app/models/katello/content_view.rb

```diff
@@ -59,6 +59,16 @@ class ContentView < Katello::Model
     validate :check_docker_conflicts
     validate :check_non_composite_auto_publish
     validates :composite, :inclusion => [true, false]
+    validates_associated :content_view_components, :message => lambda { |_view, component_attr|
+      message = ""
+      component_attr[:value].each do |component|
+        if component.errors.full_messages != []
+          message << " Content View ID #" + component.content_view_id.to_s + ": "
```

Comment:
I like this idea of identifying which content view ID is already present in the composite through another component. At the same time I think it illustrates a shortcoming of the current message in the component: "Duplicate Content View. Another component already includes this view" which is somewhat redundant to begin with (imo).

By clarifying that message you can avoid having to it here, making the solution more generic. An example might be: "Another component already includes content view with ID 4"

Your validation here might become:

```
# grab errors from all components
messages = component_attr[:value].collect { |component| component.errors.full_messages }

# messages is an array of arrays [ [ error_message_1, error_message_2], [error_message_3]]
# .flatten makes it look like [error_message_1, error_message_2, error_message_3]
# .uniq will remove any duplicate errors (if that's what we want to do)
messages = messages.flatten.uniq
return if messages.empty?
messages.join(",")
```